intermediate/tcp-client-server-multiplex/src/main/java/org/springframework/integration/samples/tcpclientserver/EchoService.java

```diff
@@ -25,11 +27,19 @@
  */
 public class EchoService {
 
-	public String test(String input) {
+	public String test(String input) throws InterruptedException {
 		if ("FAIL".equals(input)) {
 			throw new RuntimeException("Failure Demonstration");
 		}
+		else if("TIMEOUT_TEST".equals(input)){
+			Thread.sleep(3000);
+		}
+
 		return input + ":echo";
 	}
 
-}
\ No newline at end of file
+	public MessageTimeoutException noResponse(String input) {
+		return new MessageTimeoutException("No response received for " + input);
```

Comment:
Good polishing and I'm merging it anyway.
Just want to share here my five cents.

This trick `return to gateway where it will be thrown` isn't obvious and enough complicated to determine throughout the source code. 

I wonder if should say anything on the matter in the Reference Manual...